### PR TITLE
jacobian access

### LINKED ISSUE #143: Add access to Jacobian function
### Discussed in https://github.com/NREL-SIIP/PowerSimulationsDynamics.jl/discussions/140

<div type='discussions-op-text'>

<sup>Originally posted by **cuihantao** October  2, 2021</sup>
Is there any example cases for obtaining the analytical Jacobians?

I'm interested but was not able to get the SMIB case working. When creating the perturbation, the code in the Quick Start Guide shows an error (also in the docs). 

Thanks,
Hantao</div>

### PR DESCRIPTION
**Pull request by jd-lara**
closes #143 

This function now allows to get the Jacobian function and disentangles the small signal analysis from the simulation 

cc. @cuihantao this is the feature you requested.

### CODE REVIEW

File: src/base/jacobian.jl [hunk 1]
```diff
@@ -2,17 +2,58 @@
...
+    F,
+    T <: Union{Matrix{Float64}, SparseArrays.SparseMatrixCSC{Float64, Int64}},
+} <: Function
     Jf::F
-    Jv::SparseArrays.SparseMatrixCSC{Float64, Int64}
+    Jv::T
     x::Vector{Float64}
 end
 
+# This version of the function is type unstable should only be used for non-critial
```
Comment (rodrigomha) [inline src/base/jacobian.jl:14]: For non-critical aspects?

Comment (jd-lara) [reply src/base/jacobian.jl]: ops..

**Review by rodrigomha — COMMENTED** [review]
We should add later documentation on how to use this feature for different initial conditions.

**Comment by jd-lara** [comment]
Yeah, we need to add docs to this later

**Comment by cuihantao** [comment]
Thanks!!

**Review by rodrigomha — APPROVED** [review]
I'll add an issue to not forget about this docs

**Pull request merged**